src/me.rs

```diff
@@ -609,11 +669,72 @@ fn estimate_motion<T: Pixel>(
     );
 
     // Scale motion vectors to full res size
-    results.mv = MotionVector {
-      col: results.mv.col << ssdec,
-      row: results.mv.row << ssdec,
+    results.mv = results.mv << ssdec;
+
+    Some(results)
+  } else {
+    None
+  }
+}
+
+/// Refine motion estimation that was computed one level of subsampling up.
+fn refine_subsampled_motion_estimate<T: Pixel>(
+  fi: &FrameInvariants<T>, ts: &TileStateMut<'_, T>, w: usize, h: usize,
+  tile_bo: TileBlockOffset, ref_frame: RefType, ssdec: u8,
+) -> Option<FullpelSearchResult> {
+  if let Some(ref rec) =
+    fi.rec_buffer.frames[fi.ref_frames[ref_frame.to_index()] as usize]
+  {
+    let frame_bo = ts.to_frame_block_offset(tile_bo);
+    let (mvx_min, mvx_max, mvy_min, mvy_max) =
+      get_mv_range(fi.w_in_b, fi.h_in_b, frame_bo, w << ssdec, h << ssdec);
+
+    let pmv = [MotionVector { row: 0, col: 0 }; 2];
+
+    // TODO: Move lambda setup elsewhere
+    // 0.5 and 0.125 are a fudge factors
+    let lambda = (fi.me_lambda * 256.0 / (1 << (2 * ssdec)) as f64
```

Comment:
This formula is copied from somewhere else, right? Is there a way we can refactor it into a reusable function instead?

Reply:
Yeah, it makes sense to do that.